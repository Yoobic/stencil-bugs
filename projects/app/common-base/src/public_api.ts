export { AppCommonBaseModule } from './lib/app-common-base.module';

export * from './lib/bases/app-chat-page/app-chat-page.component';
export * from './lib/bases/app-entity-action-page/app-entity-action-page.component';
export * from './lib/bases/app-form-dynamic-page/app-form-dynamic-page.component';
export * from './lib/bases/app-grid-dialog-page/app-grid-dialog-page.component';
export * from './lib/bases/app-menu-page/app-menu-page.component';
export * from './lib/bases/app-root/app-root.component';
export * from './lib/bases/app-search-page/app-search-page.component';
export * from './lib/bases/app-walkthrough-page/app-walkthrough-page.component';
export * from './lib/bases/app-whatsnew-page/app-whatsnew-page.component';
export * from './lib/bases/base-page/base-page.component';

export * from './lib/components/app-profile/app-profile.component';
export * from './lib/components/app-login/app-login.component';
export * from './lib/components/app-grid/app-grid.component';
export * from './lib/components/app-form-dynamic/app-form-dynamic.component';

export * from './lib/services/dialog/dialog.service';
export * from './lib/services/utils/utils.service';

export * from './lib/animations/animations';
export * from './lib/animations/transitions';

export * from './lib/modals/app-result-dialog-page/app-result-dialog-page.component';