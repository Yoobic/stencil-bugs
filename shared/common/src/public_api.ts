export { CommonModule } from './lib/common.module';

export * from './lib/decorators/mixin/mixin.decorator';
export * from './lib/decorators/persistent/persistent.decorator';
export * from './lib/directives/long-press/long-press.directive';

export * from './lib/interfaces/animate-state/animate-state.class';
export * from './lib/interfaces/animations/animations.class';
export * from './lib/interfaces/colors/colors.interface';
export * from './lib/interfaces/config-constants/config-constants.interface';
export * from './lib/interfaces/position/position.interface';

export * from './lib/pipes/array-from-number/array-from-number.pipe';
export * from './lib/pipes/cloudinary/cloudinary.pipe';
export * from './lib/pipes/date-format/date-format.pipe';
export * from './lib/pipes/file-size/file-size.pipe';
export * from './lib/pipes/highlight/highlight.pipe';
export * from './lib/pipes/https/https.pipe';
export * from './lib/pipes/keys/keys.pipe';
export * from './lib/pipes/limit-to/limit-to.pipe';
export * from './lib/pipes/map-values/map-values.pipe';
export * from './lib/pipes/newline/newline.pipe';
export * from './lib/pipes/nohtml/nohtml.pipe';
export * from './lib/pipes/order-by/order-by.pipe';
export * from './lib/pipes/round/round.pipe';
export * from './lib/pipes/safe/safe.pipe';
export * from './lib/pipes/take/take.pipe';
export * from './lib/pipes/time-ago/time-ago.pipe';
export * from './lib/pipes/timer/timer.pipe';

export * from './lib/services/config-constants/config-constants.service';
export * from './lib/services/core-config/core-config.service';
export * from './lib/services/dom-adapter/dom-adapter.service';
export * from './lib/services/exif-restorer/exif-restorer.service';
export * from './lib/services/geo-location/geo-location.service';
export * from './lib/services/local-forage/local-forage.service';
export * from './lib/services/local-storage/local-storage.service';
export * from './lib/services/log/log.service';
export * from './lib/services/network/network.service';
export * from './lib/services/promise/promise.service';
export * from './lib/services/resize/resize.service';
export * from './lib/services/ruler/ruler.service';
export * from './lib/services/utils/utils.service';

export * from './lib/mixins/lodash-mixins/lodash-mixins';