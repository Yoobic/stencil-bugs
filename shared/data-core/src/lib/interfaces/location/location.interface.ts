import { IProperties } from '../entity/entity.interface';
import { FormFieldType, ILocation } from '@shared/interfaces';
import { LocationType } from '../location-type/location-type.interface';
import { Model } from '../../decorators/model/model.decorator';
import { Searchable } from '../../decorators/searchable/searchable.decorator';
import { Editable } from '../../decorators/editable/editable.decorator';
import { Mapping } from '../../decorators/mapping/mapping.decorator';
import { ROLES_CONDITIONS } from '../condition/condition.interface';
import { Tenant } from '../tenant/tenant.interface';

export const LOCATION_GEOCODESTATUS = ['file', 'geocoder', 'place', 'prediction', 'error'];

export function onAddressChange(value, controls, data: Location, field) {
    if (value && value._geoloc) {
        data._geoloc = value._geoloc;
        (<any>field)._geoloc = value._geoloc;
        //data.address = value.address;
        setTimeout(() => {
            controls.address.setValue(value.address, { onlySelf: true, emitModelToViewChange: false, emitEvent: false });
        });
    }
}

export function onTypeChange(value: LocationType, controls, data: Location) {
    if (value && value._id) {
        data.typeRef = value._id;
    }
}

@Model({
    className: 'Location',
    collectionName: 'locations',
    fields: ['*', 'type.name'],
    include: ['type', '_tenant']
})
export class Location implements ILocation {

    @Mapping('Location', { order: 12 })
    @Editable('Location', { type: FormFieldType.text, visible: false, forceExport: true, exportOrder: 13, filterable: false })
    _id: string;

    @Mapping('Location', { order: 11 })
    @Editable('Location', { type: FormFieldType.text, condition: ROLES_CONDITIONS.isNotTrial, columnDefinition: { width: 100 }, exportOrder: 12 })
    @Searchable('Location') clientid: string;

    @Mapping('Location', { order: 0 })
    @Editable('Location', { required: true, type: FormFieldType.text, exportOrder: 1 })
    @Searchable('Location') title: string;

    @Mapping('Location', { order: 1 })
    @Editable('Location', { required: true, type: FormFieldType.address, filterName: '_geoloc', columnDefinition: { width: 350 }, showMap: true, exportOrder: 2, onChange: onAddressChange })
    @Searchable('Location')
    address: string;

    @Mapping('Location', { order: 13 })
    @Editable('Location', { type: FormFieldType.checkbox, condition: ROLES_CONDITIONS.isNotTrial, columnDefinition: { width: 40 }, exportOrder: 14 })
    vip: boolean;

    @Mapping('Location', { order: 9 })
    @Editable('Location', { title: 'LOCATIONTAGS', condition: ROLES_CONDITIONS.isNotTrial, type: FormFieldType.autocomplete, tag: true, collectionName: 'locations', multiple: true, subQuery: { field: 'locationRef', values: '_id' }, icon: 'yo-tag', exportOrder: 10 })
    tags: Array<string>;

    @Mapping('Location', { order: 2 })
    @Editable('Location', { type: FormFieldType.autocomplete, collectionName: 'locationtypes', required: true, condition: ROLES_CONDITIONS.isNotTrial, columnDefinition: { name: 'name' }, exportOrder: 3, filterableAdvanced: true, onChange: onTypeChange })
    type: LocationType;
    typeRef: string;

    @Mapping('Location', { order: 14 })
    @Editable('Location', {
        type: FormFieldType.autocomplete, condition: ROLES_CONDITIONS.isNotTrial, collectionName: 'missiondescriptions', clearable: true, multiple: true, filterable: false, sortable: false, suppressExport: true,
        filters: [
            [{ field: 'archived', operator: { _id: 'neq' }, value: true }]
        ],
        hiddenFields: ['archived']
    })
    missiondescriptions;
    missiondescriptionsRef: Array<string>;

    @Mapping('Location', { order: 10 })
    @Editable('Location', { title: 'NOTIFICATIONEMAILS', type: FormFieldType.emailreport, condition: ROLES_CONDITIONS.isNotTrial, showUsers: true, stateful: false, tab: 'INFORMATION', exportOrder: 11, filterableAdvanced: true })
    notificationemail: Array<string>;

    @Mapping('Location', { order: 5 })
    @Editable('Location', { type: FormFieldType.text, tab: 'INFORMATION', exportOrder: 6, filterableAdvanced: true })
    @Searchable('Location') contactname: string;

    @Mapping('Location', { order: 6 })
    @Editable('Location', { type: FormFieldType.email, tab: 'INFORMATION', exportOrder: 7, filterableAdvanced: true })
    contactemail: string;

    @Mapping('Location', { order: 7 })
    @Editable('Location', { type: FormFieldType.text, tab: 'INFORMATION', exportOrder: 8, filterableAdvanced: true })
    contactphone: string;

    @Mapping('Location', { order: 8 })
    @Editable('Location', { type: FormFieldType.textarea, tab: 'INFORMATION', exportOrder: 9, filterableAdvanced: true })
    info: string;

    @Editable('Location', { readonly: true, type: FormFieldType.autocomplete, values: LOCATION_GEOCODESTATUS, clearable: false, filterableAdvanced: true, suppressExport: true })
    status: string;

    @Mapping('Location', { order: 4 })
    @Editable('Location', { visible: false, exportOrder: 5, forceExport: true, filterable: false, sortable: false })
    placesearch: string;

    @Mapping('Location', { order: 3 })
    @Editable('Location', { visible: false, forceExport: true, exportOrder: 4, filterable: false, sortable: false })
    _geoloc: [number, number];

    @Editable('Location', { type: FormFieldType.json, tab: 'PROPERTIES', filterable: false, sortable: false, suppressExport: true, condition: ROLES_CONDITIONS.isAdmin })
    properties: IProperties;

    distance: number;
    stats: Array<{ title: string; value: number; color: string }>;

    @Editable('Location', { required: true, title: 'TENANT', type: FormFieldType.autocomplete, condition: [ROLES_CONDITIONS.isAdmin], collectionName: 'tenants', multiple: false, columnDefinition: { name: 'name' } })
    _tenant: Tenant;
    _tenantRef?: string;
}
