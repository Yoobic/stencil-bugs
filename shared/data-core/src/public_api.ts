export { DataCoreModule } from './lib/data-core.module';

export * from './lib/decorators/editable/editable.decorator';
export * from './lib/decorators/mapping/mapping.decorator';
export * from './lib/decorators/model/model.decorator';
export * from './lib/decorators/searchable/searchable.decorator';

export * from './lib/interfaces/aggregate-log/aggregate-log.interface';
export * from './lib/interfaces/algorithm/algorithm.interface';
export * from './lib/interfaces/autorenew-config/autorenew-config.interface';
export * from './lib/interfaces/backup/backup.interface';
export * from './lib/interfaces/catalog/catalog.interface';
export * from './lib/interfaces/chart-definition/chart-definition.interface';
export * from './lib/interfaces/condition/condition.interface';
export * from './lib/interfaces/condition/icondition.interface';
export * from './lib/interfaces/constants/constants.interface';
export * from './lib/interfaces/currency/currency.interface';
export * from './lib/interfaces/dashboard/dashboard.interface';
export * from './lib/interfaces/entity/entity.interface';
export * from './lib/interfaces/field-selector/field-selector.interface';
export * from './lib/interfaces/file/file.interface';
export * from './lib/interfaces/folder/folder.interface';
export * from './lib/interfaces/group/group.interface';
export * from './lib/interfaces/kpi/kpi.interface';
export * from './lib/interfaces/location-type/location-type.interface';
export * from './lib/interfaces/location/location.interface';
export * from './lib/interfaces/mapping/mapping.interface';
export * from './lib/interfaces/mission-description/mission-description.interface';
export * from './lib/interfaces/model/model.interface';
export * from './lib/interfaces/notification-event/notification-event.interface';
export * from './lib/interfaces/notification/notification.interface';
export * from './lib/interfaces/operation/operation.interface';
export * from './lib/interfaces/product/product.interface';
export * from './lib/interfaces/response-object/response-object.interface';
export * from './lib/interfaces/slide/slide.interface';
export * from './lib/interfaces/sort/sort.interface';
export * from './lib/interfaces/tableau/tableau.interface';
export * from './lib/interfaces/tag-group/tag-group.interface';
export * from './lib/interfaces/tag/tag.interface';
export * from './lib/interfaces/tenant/tenant.interface';
export * from './lib/interfaces/todo/todo.interface';
export * from './lib/interfaces/translation/translation.interface';
export * from './lib/interfaces/user/user.interface';
export * from './lib/interfaces/workplace/workplace.interface';

export * from './lib/pipes/authorized/authorized.pipe';
export * from './lib/pipes/currency/currency.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/username/username.pipe';
export * from './lib/pipes/useremail/useremail.pipe';
export * from './lib/pipes/userinitial/userinitial.pipe';

export * from './lib/services/activity/activity.service';
export * from './lib/services/algorithms/algorithms.service';
export * from './lib/services/authentication/authentication.guard';
export * from './lib/services/authentication/authentication.service';
export * from './lib/services/authentication/current-session.resolver';
export * from './lib/services/backup/backup.service';
export * from './lib/services/box/box.service';
export * from './lib/services/blog/blog.service';
export * from './lib/services/links/links.service';
export * from './lib/services/broker/broker.service';
export * from './lib/services/config/config.service';
export { Dashboard as DashboardCore } from './lib/services/dashboard/dashboard.service';
export * from './lib/services/data-loader/data-loader.service';
export * from './lib/services/files/files.service';
export * from './lib/services/files/files-broker.service';
export * from './lib/services/googlemaps/googlemaps.service';
export * from './lib/services/googletranslate/googletranslate.service';
export * from './lib/services/http-interceptor/http-interceptor.service';
export * from './lib/services/loading-bar/loading-bar.service';
export * from './lib/services/locations/locations.service';
export * from './lib/services/loopback/loopback.service';
export * from './lib/services/models/models.service';
export * from './lib/services/print/print.service';
export * from './lib/services/push/push.service';
export * from './lib/services/requestor/requestor.service';
export * from './lib/services/security/security.service';
export * from './lib/services/session/session.service';
export * from './lib/services/smartloc/smartloc.service';
export * from './lib/services/users/users.service';
export * from './lib/services/version/version.service';
export * from './lib/services/workplace/workplace.service';
export * from './lib/services/xlsx/xlsx.service';

export * from './lib/validators/password-validator/password-validator';
export * from './lib/validators/username-validator/username-validator';
