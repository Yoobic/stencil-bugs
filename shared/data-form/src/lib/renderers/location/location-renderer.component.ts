import { Component } from '@angular/core';
import { ICellRendererAngularComp } from 'ag-grid-angular';
import { DefaultRendererComponent } from '../default/default-renderer.component';

@Component({
    selector: 'location-cell',
    template: `{{value()}}`
})
export class LocationRendererComponent implements ICellRendererAngularComp {
    private params: any;

    agInit(params: any) {
        this.params = params;
    }

    value() {
        let value = DefaultRendererComponent.renderer(this.params);
        return value ? (value.title ? value.title + ' - ' : '') + value.address : '';
    }

    refresh(params: any) {
        return false;
    }
}