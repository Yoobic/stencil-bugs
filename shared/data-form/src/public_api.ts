
export { DataFormModule } from './lib/data-form.module';

export * from './lib/interfaces/custom-form-field/custom-form-field.interface';
export * from './lib/interfaces/form-field/form-field.class';
export * from './lib/interfaces/gridstate/gridstate.interface';
export * from './lib/interfaces/cache/cache.interface';
export * from './lib/interfaces/custom-model/custom-model.interface';

export * from './lib/services/cell-renderer/cell-renderer.service';
export * from './lib/services/chart/chart.service';
export * from './lib/services/form-dynamic-builder/form-dynamic-builder.service';
export * from './lib/services/conditions/conditions.service';
export * from './lib/services/mapping/mapping.service';
export * from './lib/services/custom-models/custom-models.service';

export * from './lib/validators/form-filter-field-validator/form-filter-field-validator';
export * from './lib/validators/form-input-base-email-validator/form-input-base-email-validator';
export * from './lib/validators/form-input-base-number-validator/form-input-base-number-validator';
export * from './lib/validators/form-input-between-validator/form-input-between-validator';
export * from './lib/validators/form-input-checklist-validator/form-input-checklist-validator';
export * from './lib/validators/form-input-missionfield-validator/form-input-missionfield-validator';
export * from './lib/validators/form-input-multiphotos-validator/form-input-multiphotos-validator';
export * from './lib/validators/form-input-photo-validator/form-input-photo-validator';
export * from './lib/validators/form-input-select-validator/form-input-select-validator';
export * from './lib/validators/password-are-not-matching-validator/password-are-not-matching-validator';

export * from './lib/renderers/address/address-renderer.component';
export * from './lib/renderers/boolean/boolean-renderer.component';
export * from './lib/renderers/button/button-renderer.component';
export * from './lib/renderers/checklist/checklist-renderer.component';
export * from './lib/renderers/conformity-progress/conformity-progress-renderer.component';
export * from './lib/renderers/conformity-relative-progress/conformity-relative-progress-renderer.component';
export * from './lib/renderers/custom-model/custom-model-renderer.component';
export * from './lib/renderers/datetime/datetime-renderer.component';
export * from './lib/renderers/default/default-renderer.component';
export * from './lib/renderers/delta/delta-renderer.component';
export * from './lib/renderers/distance/distance-renderer.component';
export * from './lib/renderers/duration/duration-renderer.component';
export * from './lib/renderers/location/location-renderer.component';
export * from './lib/renderers/mission-status/mission-status-renderer.component';
export * from './lib/renderers/mission-validation/mission-validation-renderer.component';
export * from './lib/renderers/number/number-renderer.component';
export * from './lib/renderers/multiphotos/multiphotos-renderer.component';
export * from './lib/renderers/percentage/percentage-renderer.component';
export * from './lib/renderers/photo/photo-renderer.component';
export * from './lib/renderers/progress/progress-renderer.component';
export * from './lib/renderers/time/time-renderer.component';
export * from './lib/renderers/timer/timer-renderer.component';
export * from './lib/renderers/todo/todo-renderer.component';
export * from './lib/renderers/user-full-name/user-full-name-renderer.component';
export * from './lib/renderers/validation-progress/validation-progress-renderer.component';
export * from './lib/renderers/video/video-renderer.component';