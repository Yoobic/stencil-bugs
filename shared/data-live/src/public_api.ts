
export { DataLiveModule } from './lib/data-live.module';

export { Channel as ChannelInterface } from './lib/interfaces/channel/channel.interface';
export * from './lib/interfaces/channels/channels.interface';
export * from './lib/interfaces/message/message.interface';
export * from './lib/interfaces/message/message.model';

export * from './lib/services/channel/channel.service';
export * from './lib/services/intercom/intercom.service';
export * from './lib/services/messages/messages.service';
export * from './lib/services/pubnub/pubnub.service';
export * from './lib/services/track/track.service';
export * from './lib/services/twilio-token/twilio-token.service';
export * from './lib/services/raven-error-handler/raven-error-handler.service';