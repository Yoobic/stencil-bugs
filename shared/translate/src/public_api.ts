export { TranslateModule } from './lib/translate.module';

export * from './lib/pipes/distance/distance.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/polyglot/polyglot.pipe';

export * from './lib/services/translate/translate.service';