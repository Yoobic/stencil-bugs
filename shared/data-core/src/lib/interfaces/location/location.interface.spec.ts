import { Location } from './location.interface';
import { Models } from '../../../public_api';

describe('shared: data-core: interfaces: Location', () => {
    it('should create an entity', () => {
        let entity = new Location();
        expect(entity._id).toBeUndefined();
        let model = Models.getModel('Location');
        expect(model).toBeTruthy();
        expect(model).toMatchSnapshot();
    });
});