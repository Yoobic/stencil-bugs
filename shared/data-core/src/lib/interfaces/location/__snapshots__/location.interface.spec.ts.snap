// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared: data-core: interfaces: Location should create an entity 1`] = `
IModel {
  "_formFields": Array [
    Object {
      "exportOrder": 13,
      "filterName": "_id",
      "filterable": false,
      "forceExport": true,
      "name": "_id",
      "title": "_ID",
      "type": "text",
      "visible": false,
    },
    Object {
      "columnDefinition": Object {
        "width": 100,
      },
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "exportOrder": 12,
      "filterName": "clientid",
      "name": "clientid",
      "title": "CLIENTID",
      "type": "text",
    },
    Object {
      "exportOrder": 1,
      "filterName": "title",
      "name": "title",
      "required": true,
      "title": "TITLE",
      "type": "text",
    },
    Object {
      "columnDefinition": Object {
        "width": 350,
      },
      "exportOrder": 2,
      "filterName": "_geoloc",
      "name": "address",
      "onChange": [Function],
      "required": true,
      "showMap": true,
      "title": "ADDRESS",
      "type": "address",
    },
    Object {
      "columnDefinition": Object {
        "width": 40,
      },
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "exportOrder": 14,
      "filterName": "vip",
      "name": "vip",
      "title": "VIP",
      "type": "checkbox",
    },
    Object {
      "collectionName": "locations",
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "exportOrder": 10,
      "filterName": "tags",
      "icon": "yo-tag",
      "multiple": true,
      "name": "tags",
      "subQuery": Object {
        "field": "locationRef",
        "values": "_id",
      },
      "tag": true,
      "title": "LOCATIONTAGS",
      "type": "autocomplete",
    },
    Object {
      "collectionName": "locationtypes",
      "columnDefinition": Object {
        "name": "name",
      },
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "exportOrder": 3,
      "filterName": "typeRef",
      "filterableAdvanced": true,
      "name": "type",
      "onChange": [Function],
      "required": true,
      "title": "TYPE",
      "type": "autocomplete",
    },
    Object {
      "clearable": true,
      "collectionName": "missiondescriptions",
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "filterName": "missiondescriptionsRef",
      "filterable": false,
      "filters": Array [
        Array [
          Object {
            "field": "archived",
            "operator": Object {
              "_id": "neq",
            },
            "value": true,
          },
        ],
      ],
      "hiddenFields": Array [
        "archived",
      ],
      "multiple": true,
      "name": "missiondescriptions",
      "sortable": false,
      "suppressExport": true,
      "title": "MISSIONDESCRIPTIONS",
      "type": "autocomplete",
    },
    Object {
      "condition": Object {
        "group": Array [
          "trial",
        ],
        "operator": "nin",
        "type": "roles",
      },
      "exportOrder": 11,
      "filterName": "notificationemail",
      "filterableAdvanced": true,
      "name": "notificationemail",
      "showUsers": true,
      "stateful": false,
      "tab": "INFORMATION",
      "title": "NOTIFICATIONEMAILS",
      "type": "emailreport",
    },
    Object {
      "exportOrder": 6,
      "filterName": "contactname",
      "filterableAdvanced": true,
      "name": "contactname",
      "tab": "INFORMATION",
      "title": "CONTACTNAME",
      "type": "text",
    },
    Object {
      "exportOrder": 7,
      "filterName": "contactemail",
      "filterableAdvanced": true,
      "name": "contactemail",
      "tab": "INFORMATION",
      "title": "CONTACTEMAIL",
      "type": "email",
    },
    Object {
      "exportOrder": 8,
      "filterName": "contactphone",
      "filterableAdvanced": true,
      "name": "contactphone",
      "tab": "INFORMATION",
      "title": "CONTACTPHONE",
      "type": "text",
    },
    Object {
      "exportOrder": 9,
      "filterName": "info",
      "filterableAdvanced": true,
      "name": "info",
      "tab": "INFORMATION",
      "title": "INFO",
      "type": "textarea",
    },
    Object {
      "clearable": false,
      "filterName": "status",
      "filterableAdvanced": true,
      "name": "status",
      "readonly": true,
      "suppressExport": true,
      "title": "STATUS",
      "type": "autocomplete",
      "values": Array [
        "file",
        "geocoder",
        "place",
        "prediction",
        "error",
      ],
    },
    Object {
      "exportOrder": 5,
      "filterName": "placesearch",
      "filterable": false,
      "forceExport": true,
      "name": "placesearch",
      "sortable": false,
      "title": "PLACESEARCH",
      "visible": false,
    },
    Object {
      "exportOrder": 4,
      "filterName": "_geoloc",
      "filterable": false,
      "forceExport": true,
      "name": "_geoloc",
      "sortable": false,
      "title": "_GEOLOC",
      "visible": false,
    },
    Object {
      "condition": Object {
        "group": Array [
          "admin",
        ],
        "operator": "in",
        "type": "roles",
      },
      "filterName": "properties",
      "filterable": false,
      "name": "properties",
      "sortable": false,
      "suppressExport": true,
      "tab": "PROPERTIES",
      "title": "PROPERTIES",
      "type": "json",
    },
  ],
  "appareance": Map {},
  "className": "Location",
  "collectionName": "locations",
  "feathersService": undefined,
  "fields": Array [
    "*",
    "type.name",
  ],
  "include": Array [
    "type",
  ],
  "isCustom": undefined,
  "mappingFields": Map {
    12 => "_id",
    11 => "clientid",
    0 => "title",
    1 => "address",
    13 => "vip",
    9 => "tags",
    2 => "type",
    14 => "missiondescriptions",
    10 => "notificationemail",
    5 => "contactname",
    6 => "contactemail",
    7 => "contactphone",
    8 => "info",
    4 => "placesearch",
    3 => "_geoloc",
  },
  "searchSubquery": undefined,
  "searchableFields": Array [
    "clientid",
    "title",
    "address",
    "contactname",
  ],
  "type": [Function],
}
`;
